lib/internal/modules/esm/resolve.js

```diff
@@ -924,29 +925,36 @@ function resolveAsCommonJS(specifier, parentURL) {
     // If it is a relative specifier return the relative path
     // to the parent
     if (isRelativeSpecifier(specifier)) {
-      found = relative(parent, found);
-      // Add '.separator if the path does not start with '..separator'
+      const foundURL = pathToFileURL(found).pathname;
+      found = relativePosixPath(
+        StringPrototypeSlice(parentURL, 'file://'.length, StringPrototypeLastIndexOf(parentURL, '/')),
```

Comment:
Why not go directly for `url.pathname`?

```suggestion
        StringPrototypeSlice(parentURL.pathname, 0, StringPrototypeLastIndexOf(parentURL.pathname, '/')),
```

Reply:
To avoid the `new URL` call